docs/r2/migrate.ipynb

```diff
@@ -0,0 +1,404 @@
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "em7GQju5VA0I"
+      },
+      "source": [
+        "\n",
+        "Example usage with eager execution, the default in TF 2.0:"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "colab_type": "code",
+        "id": "GgFXOtSeVFqP",
+        "colab": {}
+      },
+      "source": [
+        "writer = tf.summary.create_file_writer(\"/tmp/mylogs/eager\")\n",
+        "\n",
+        "with writer.as_default():\n",
+        "  for step in range(100):\n",
+        "    # other model code would go here\n",
+        "    tf.summary.scalar(\"my_metric\", 0.5, step=step)\n",
+        "    \n",
+        "writer.flush()"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "colab_type": "code",
+        "id": "h5fk_NG7QKve",
+        "colab": {}
+      },
+      "source": [
+        "ls /tmp/mylogs/eager"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "FvBBeFxZVLzW"
+      },
+      "source": [
+        "\n",
+        "Example usage with tf.function graph execution:"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "colab_type": "code",
+        "id": "kovK0LEEVKjR",
+        "colab": {}
+      },
+      "source": [
+        "writer = tf.summary.create_file_writer(\"/tmp/mylogs/tf_function\")\n",
+        "\n",
+        "@tf.function\n",
+        "def my_func(step):\n",
+        "  with writer.as_default():\n",
+        "    # other model code would go here\n",
+        "    tf.summary.scalar(\"my_metric\", 0.5, step=step)\n",
+        "\n",
+        "for step in range(100):\n",
+        "  my_func(step)\n",
+        "  writer.flush()"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "colab_type": "code",
+        "id": "Qw5nHhRUSM7_",
+        "colab": {}
+      },
+      "source": [
+        "ls /tmp/mylogs/tf_function"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "5SY6eYitUJH_"
+      },
+      "source": [
+        "\n",
+        "\n",
+        "\n",
+        "\n",
+        "\n",
+        "Example usage with legacy TF 1.x graph execution:\n",
+        "\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "colab_type": "code",
+        "id": "OyQgeqZhVRNB",
+        "colab": {}
+      },
+      "source": [
+        "with tf.compat.v1.Graph().as_default() as g:\n",
```

Comment:
It's a little weird to me to use the face that `g` exists as a local name outside the `with` block (I guess this is python's behavior, it was new to me though).  Maybe more obvious as 

```python
g = tf.compat.v1.Graph()
with g.as_default():
  ...
with tf.compat.v1.Session(graph=g) as sess:
  ...
```

Reply:
Yes. that does seem to surprise everyone the first time.